### ISSUE TITLE
azure_openai set json_schema Error: Completions.create() got an unexpected keyword argument 'json_schema'

### ISSUE THREAD

**Issue by jakcm** [langgenius/dify#8451]
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.8.2

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

1. Use azure openai service, register model gpt-4o-2024-08-06, and use API version 2024-08-01-preview.

2. response_format Select json_schema and fill in JSON Schema: 
{
	"name": "json_01",
	"description": "Analyze emotions",
	"strict": true,
	"schema": {
		"type": "object",
		"properties": {
			"emotion": {
				"type": "string",
				"description": "Analyze emotions by user input"
			}
		},
		"required": ["emotion"],
		"additionalProperties": false
	}
}

4. Run input: Good job. Run failed: [azure_openai] Error: Completions.create() got an unexpected keyword argument 'json_schema'
 

### ✔️ Expected Behavior

Output JSON.

### ❌ Actual Behavior

Run failed: [azure_openai] Error: Completions.create() got an unexpected keyword argument 'json_schema'

**Comment by tuoxiansp** [comment]
I encountered the same issue. After debugging, I found that in the code of the Azure model provider, the received response_format parameter is None. This results in the subsequent code for handling json_schema not being executed.
@hjlarry 

```python

    def _chat_generate(
        self,
        model: str,
        credentials: dict,
        prompt_messages: list[PromptMessage],
        model_parameters: dict,
        tools: Optional[list[PromptMessageTool]] = None,
        stop: Optional[list[str]] = None,
        stream: bool = True,
        user: Optional[str] = None,
    ) -> Union[LLMResult, Generator]:
        client = AzureOpenAI(**self._to_credential_kwargs(credentials))

        # response_format = None
        response_format = model_parameters.get("response_format")
        if response_format:
            if response_format == "json_schema":
                json_schema = model_parameters.get("json_schema")
                if not json_schema:
                    raise ValueError("Must define JSON Schema when the response format is json_schema")
                try:
                    schema = json.loads(json_schema)
                except:
                    raise ValueError(f"not correct json_schema format: {json_schema}")
                model_parameters.pop("json_schema")
                model_parameters["response_format"] = {"type": "json_schema", "json_schema": schema}
            else:
                model_parameters["response_format"] = {"type": response_format}

```

https://github.com/langgenius/dify/blob/b279d1904085d0265c4b0a0ef5843bc11b22c837/api/core/model_runtime/model_providers/azure_openai/llm/llm.py#L284C9-L284C24

**Comment by hjlarry** [comment]
@tuoxiansp thanks for debugging.  I've fix it in https://github.com/langgenius/dify/pull/9148 ,  but I haven't got a AZURE key, can you help to pull it and try if it is worked?

**Comment by tuoxiansp** [comment]
> @tuoxiansp thanks for debugging. I've fix it in #9148 , but I haven't got a AZURE key, can you help to pull it and try if it is worked?

I've tested it, and it works perfectly. Thank you for fixing it so quickly!
